### DISCUSSION TITLE
Getting lots of compilation errors settting up DBeaver in Eclipse

### DISCUSSION THREAD

**Discussion by RedTailedHawk** [dbeaver/dbeaver#9622] [DBeaver Development] [answered]
Just cloned the project and imported into Eclipse via pom.xml files, but getting lots of compilation errors.

I'm a newbie to Eclipse RCP, so perhaps I'm missing a plugin?

Thanks for any help.

![image](https://user-images.githubusercontent.com/8505928/91212671-895e2c80-e6de-11ea-8e53-3d3d651721b1.png)

**Comment by RedTailedHawk** [comment]
FYI, I just installed the Eclipse SDK, but still getting build errors.

![image](https://user-images.githubusercontent.com/8505928/91221095-2a52e480-e6eb-11ea-9a29-f25501e3bdd5.png)

**Comment by RedTailedHawk** [comment]
FWIW, I reinstalled Eclipse as `Eclipse IDE for RCP and RAP Developers` and rebuilt the workspace, but still getting 4200+ compilation errors.

**Comment by uslss** [comment, marked as answer]
Accordingly to step 14 from this instruction https://github.com/dbeaver/dbeaver/wiki/Develop-in-Eclipse

> If some modules build will fail (they will be marked with red cross) - update Maven project

**Comment by RedTailedHawk** [comment]
Thanks @uslss, I just needed to add those two dependencies mentioned in the Eclipse Development section.  Appreciate the help!

**Comment by RedTailedHawk** [comment]
OK, I get a clean build now, but when I try to debug the app I see the DBeaver splash screen, but then it says "The application could not start."

The Eclipse console shows a bunch of ClassNotFoundException messages.  For example:

`Caused by: java.lang.ClassNotFoundException: org.eclipse.swt.SWTError cannot be found by org.eclipse.ui.workbench_3.119.0.v20200521-1247`

I went to the Run Configuration and checked `Add Required Plug-ins`, but that didn't seem to help.

**Comment by RedTailedHawk** [comment]
Figured I would update all of the Maven projects, but that's giving lots of errors. :-(

![image](https://user-images.githubusercontent.com/8505928/91565280-f6a4d400-e90f-11ea-983b-e91c5501b8ae.png)

**Comment by RedTailedHawk** [comment]
I actually just started from scratch on this (with Eclipse 2020-09) and got it to work.  Looks like I just needed to root the workspace properly.  i.e. I needed to import from the folder *above* the one that contained the root `pom.xml`, and *not* import from the folder that contained it.

![image](https://user-images.githubusercontent.com/8505928/98684225-e9568d80-2333-11eb-961a-7b6957234b91.png)
